truss/templates/server/model_wrapper.py

```diff
@@ -307,32 +318,58 @@ def ready(self) -> bool:
     def _model_file_name(self) -> str:
         return self._config["model_class_filename"]
 
-    def start_load_thread(self):
-        # Don't retry failed loads.
-        if self._status == ModelWrapper.Status.NOT_READY:
-            thread = Thread(target=self.load)
-            thread.start()
+    async def setup_environment(self, environment: dict):
+        descriptor = self.model_descriptor.setup_environment
+        if not descriptor:
+            return
+        if descriptor.is_async:
+            return await self._model.setup_environment(environment)
+        else:
+            return await to_thread.run_sync(self._model.setup_environment, environment)
+
+    async def poll_for_environment_updates(self) -> None:
+        last_modified_time = None
+        ENVIRONMENT_CONFIG_FILENAME = "/tmp/environment"
+        SLEEP_TIME_SECONDS = 10
+        while True:
```

Comment:
ok to keep an infinite loop in an asyncio task here? do we need to worry about graceful shutdown?

Reply:
there's no problem with keeping an infinite loop in an asyncio task, your task yields at the end, so it doesn't block anything

> do we need to worry about graceful shutdown?

what do you mean?

Reply:
I'm referring to whether we need to do any special error handling in this while loop when the event loop is shutting down but since we're just polling and calling the `setup_environment` user-defined function here I think its fairly safe to not need to handle any particular errors from the event loop on shutdown